static/app/components/modals/debugFileCustomRepository/http.tsx

```diff
@@ -0,0 +1,285 @@
+            value={data.username}
+            onChange={e =>
+              setData({
+                ...data,
+                username: e.target.value,
+              })
+            }
+          />
+        </Field>
+        <Field
+          label={t('Password')}
+          inline={false}
+          help={t('Password for HTTP basic auth')}
+          flexibleControlStateSize
+          stacked
+        >
+          <PasswordInput
+            type={data.password === undefined ? 'text' : 'password'}
+            name="url"
+            placeholder={
+              data.password === undefined ? t('(Password unchanged)') : 'open-sesame'
+            }
+            value={data.password}
+            onChange={e =>
+              setData({
+                ...data,
+                password: e.target.value,
+              })
+            }
+          />
+          {(data.password === undefined ||
+            (typeof data.password === 'string' && !!data.password)) && (
+            <ClearPasswordButton
+              onClick={handleClearPassword}
+              icon={<IconClose size="14px" />}
+              size="xsmall"
+              title={t('Clear password')}
+              label={t('Clear password')}
+              borderless
+            />
+          )}
+        </Field>
+        <Divider />
+        <StyledSelectField
+          name="layout.type"
+          label={t('Directory Layout')}
+          help={t('The layout of the folder structure.')}
+          choices={Object.keys(DEBUG_SOURCE_LAYOUTS).map(key => [
+            key,
+            DEBUG_SOURCE_LAYOUTS[key],
+          ])}
+          value={data['layout.type']}
+          onChange={value =>
+            setData({
+              ...data,
+              ['layout.type']: value,
+            })
+          }
+          inline={false}
+          flexibleControlStateSize
+          stacked
+        />
+        <StyledSelectField
+          name="layout.casing"
+          label={t('Path Casing')}
+          help={t('The case of files and folders.')}
+          choices={Object.keys(DEBUG_SOURCE_CASINGS).map(key => [
+            key,
+            DEBUG_SOURCE_CASINGS[key],
+          ])}
+          value={data['layout.casing']}
+          onChange={value =>
+            setData({
+              ...data,
+              ['layout.casing']: value,
+            })
+          }
+          inline={false}
+          flexibleControlStateSize
+          stacked
+        />
+      </Body>
+      <Footer>
+        <Button
+          onClick={handleSubmit}
+          priority="primary"
+          disabled={isFormInvalid() || formUnchanged()}
+        >
+          {t('Save changes')}
+        </Button>
+      </Footer>
+    </Fragment>
+  );
+}
+
+export default Http;
+
+const StyledSelectField = styled(SelectField)`
+  padding-right: 0;
+`;
+
+const Divider = styled('div')`
+  height: 1px;
+  background-color: ${p => p.theme.gray200};
+  margin-bottom: ${space(2)};
+`;
```

Comment:
Can we use `FieldSeparator`?

Reply:
I noticed that we have the `FieldSepatator` component, but it's just the <hr /> tag. Do you think it makes sense in having this component? I can use this tag, but I'll have to style it to remove the padding-top

Reply:
I think so - it has a specific purpose (separates fields in forms), and I could see us needing to broadly style it.

Although you could also make the argument that it's not used much/at all (aside from potentially plugins via `FieldFromConfig`)

Reply:
Actually, can you post a screenshot of what the separator looks like?

Reply:
![image](https://user-images.githubusercontent.com/29228205/131524217-4e8fb661-7dc4-4f35-a639-feeb27e753ab.png)
@billyvg here is one example

Reply:
Ahh yeah, it definitely needs to be styled haha. Does your modal form look similar where there is extra `margin-top`?

I think we should use `FieldSeparator` - I can follow-up and update it to make sure it's styled properly.

Otherwise this looks good, I'll leave it to @jan-auer for final approval